### ISSUE TITLE
Changes to the A2S_INFO protocol

### ISSUE THREAD

**Issue by WinterPhoenix** [Yepoleb/python-a2s#19]
A2S_INFO, A2S_PLAYER, and A2S_RULES packets are changing to solve a reflection attack vulnerability. This project will need to be updated in order for queries to continue to work.

More info here: https://steamcommunity.com/discussions/forum/14/2989789048633291344/

Thanks for all the hard work!

**Comment by Yepoleb** [comment]
Thanks for the heads up! This shouldn't be too difficult to implement.

**Comment by Yepoleb** [comment]
I've committed this in a branch and will merge it as soon as the feature in Steam leaves beta. Currently the padding is not configurable, but I might add it for compatibility later if there is a relevant number of servers that doesn't implement the change. Official policy is still "it's not supported if Steam doesn't support it" but it might be worth it to make an exception here. https://github.com/Yepoleb/python-a2s/tree/padding

**Comment by WinterPhoenix** [comment]
Wonderful, looks like it's working, thanks!

**Comment by WinterPhoenix** [comment]
Heads up, it looks like Valve is changing the approach they're using for the A2S_INFO changes due to the padding triggering existing Anti-DDoS filters:
https://steamcommunity.com/discussions/forum/14/2989789048633291344/#c2974028351329871253

**Comment by Yepoleb** [comment]
Thanks, will implement this as soon as possible.

**Comment by Nereg** [comment]
![image](https://user-images.githubusercontent.com/47698366/101641640-aa9f1a80-3a3a-11eb-9929-04fb0bbee359.png)
it seems like they implemented it already in official client. And [here] is link mentioned in the changelogs
P.S. For now my test script and project is working fine with ARK: Survival Evolved.

**Comment by Nereg** [comment]
I found test server with new format (138.201.84.93:27113) and this library doesn't wotk with it at all. Here is stack trace:
```
Received single packet: b'A\xad $\x00'
Traceback (most recent call last):
  File "test.py", line 22, in <module>
    server = a2s.info(address)
  File "D:\Projects\Current\ARK Server monitor\lib\site-packages\a2s\info.py", line 272, in info
    return info_response(resp_data, ping, encoding)
  File "D:\Projects\Current\ARK Server monitor\lib\site-packages\a2s\info.py", line 259, in info_response
    "Invalid response type: " + str(response_type))
a2s.exceptions.BrokenMessageError: Invalid response type: 65
```
I took it from responce [here](https://steamcommunity.com/discussions/forum/14/2974028351344359625/):
```
Thanks for the information.

I successfully tested the new protocol with this configuration:
- TOXIKK game server (app id 324810)
- dropped the dlls from the zip you linked above into C:\steamcmd\steamapps\common\TOXIKK\Binaries\Win32, replacing the older DLLs that are shipped with the game
- setting STEAM_GAMESERVER_A2S_INFO_REQUIRE_CHALLENGE=1
- started the server

The Steam Client's built-in browser (View / Servers) and also the game's internal server browser did not show my server before I opted-in to the steam client beta. After beta opt-in the server was listed again in both.

My SteamServerBrowser app constructs UDP query packets directly without steamclient.dll and works now too after a bug fix. I can confirm that the server is sending a challenge as expected.

If anyone needs a server for testing, you can use mine: 138.201.84.93:27113 
```
And also this author implemented that in his client : https://github.com/PredatH0r/SteamServerBrowser

**Comment by Yepoleb** [comment]
I did not have a test server and that delayed the implementation, but with a public one I can move forward much quicker. Thanks for the hint!

**Comment by Yepoleb** [comment]
Implemented in a453675feaf577dd7faa7be26281b52ac1f8a4ef

**Comment by WinterPhoenix** [comment]
Thanks for all the hard work Yepoleb!
